k8s/upgrade/src/preflight_validations.rs

```diff
@@ -0,0 +1,20 @@
+use crate::user_prompt;
+use operators::upgrade::{common::error, controller::reconciler};
+
+/// Validation to be done before applying upgrade.
+pub async fn preflight_check() -> Result<(), error::Error> {
+    console_logger::info(user_prompt::UPGRADE_WARNING);
+    match reconciler::is_rebuilding().await {
```

Comment:
maybe split the rebuild into a separate function? So if we ever have more checks it can be more "modular"